src/cratedb_fivetran_destination/main.py

```diff
@@ -153,33 +144,36 @@ def AlterTable(self, request, context):
                 "Alter table intends to change the primary key of the table. "
                 "Because CrateDB does not support this operation, the table will be recreated.",
             )
-            # FIXME: Implement non-inplace ALTER TABLE for primary key updates.
-            """
-            ats = AlterTableRecreateStatements(
-                table=table_info, table_new=new_table, columns_common=columns_common
-            )
-            """
-            msg = (
-                "Alter table intends to change the primary key of the table, "
-                "but this operation is not implemented yet"
-            )
-            log_message(LOG_SEVERE, f"AlterTable: {msg}")
-            return destination_sdk_pb2.AlterTableResponse(
-                success=False,
-                warning=common_pb2.Warning(message=msg),
+
+            # TODO: Refactor this _into_ the `AlterTableRecreateStatements` workhorse function.
+            temptable = request.table.name + "_alter_tmp"
```

Comment:
@coderabbitai: Please carry over this proposal from your pen into a dedicated ticket, thanks.

<summary>src/cratedb_fivetran_destination/main.py (2)</summary><blockquote>

`149-155`: **Consider making temp table creation resilient to leftovers from failed runs**

When `pk_has_changed` is true, `_create_table` is called for a deterministic temp name (`<table>_alter_tmp`). If a previous migration fails between creating the temp table and dropping it, a subsequent run will see `CREATE TABLE` fail with “table already exists”. You might want either a preceding `DROP TABLE IF EXISTS` for the temp name or a more unique suffix (e.g., timestamp/UUID) to tolerate interrupted runs.

</blockquote>